src/map/lua/lua_baseentity.cpp

```diff
@@ -13475,13 +13475,21 @@ uint16 CLuaBaseEntity::getStealItem()
 
         if (PDropList && !PMob->m_ItemStolen)
         {
+            // Steal item randomly selected from steal drop table
+            std::vector<uint16> Items;
+
             for (const DropItem_t& drop : PDropList->Items)
             {
                 if (drop.DropType == DROP_STEAL)
                 {
-                    return drop.ItemID;
+                    Items.emplace_back(drop.ItemID);
                 }
             }
+
+            if (!Items.empty())
+            {
+                return Items[xirand::GetRandomNumber(Items.size())];
```

Comment:
For passers-by: get random number doesn't include the max for integer types:
https://github.com/LandSandBoat/server/blob/7525508ad051b7ba40e46576d4a37ba5ee508501/src/common/xirand.h#L56

But it does for floating types, so be careful.

This usage is fine though.

Reply:
Oh, we have a helper for this, use this instead:
https://github.com/LandSandBoat/server/blob/7525508ad051b7ba40e46576d4a37ba5ee508501/src/common/xirand.h#L95